### ISSUE TITLE
OPAM 2.0 & ocaml-version

### ISSUE THREAD

**Issue by Fourchaux** [ocaml/opam#2751]
With 'OPAM 2.0~alpha4' & 'OCaml 4.04.0+flambda'

Trying to 'opam upgrade' last version of Merlin (v2.5.2) displays : 

Upgrade is not possible because of conflicts or packages that are no longer available:
merlin.2.5.2 has unmet availability conditions: ocaml-version >= "4.02.3" & ocaml-version < "4.05.0"

Line used in (Merlin) opam file :
available: [ocaml-version >= "4.02.3" & ocaml-version < "4.05.0"]

**Note:**
No such problem with 'OCaml 4.04.0' .

**Comment by AltGr** [comment]
Make sure your repository is rewritten using `opam-admin upgrade-format` and there should be no problem.
If you pin the package and the opam file is found in its source, opam will rewrite it on the fly.

**Comment by Fourchaux** [comment]
See: https://github.com/the-lambda-church/merlin/commit/b4af57726f14761d8a557306c80a882c7755d06d

Using these constraints solve the problem. (TY @let-def)
available: [ ocaml-version >= "4.02.1" & ocaml-version < "4.05" ]

@AltGr
I don't have 'opam-admin'. This tool has a specific installation ?  

My OPAM :  pre-compiled binary file (opam-2.0-alpha5-x86_64Linux)

**Comment by AltGr** [comment]
There was an issue yesterday due to a premature change of the archive sync/caching format, that caused `opam update` to fail on pre-releases of 2.0 ; without being able to update to the rewritten 2.0 repository, opam would stay on the 1.2 repository, which has, among other things, `ocaml-version` variables it is not supposed to know about.

Tl;dr: repo server issue fixed upstream, `opam update` should get you going.
